test/unittest/unittest_common_properties.cpp

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file unittest_common_properties.h
+ * @date 15 May 2021
+ * @brief This file contains test and specification of properties and exporter
+ * @see	https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+#include <gtest/gtest.h>
+
+#include <common_properties.h>
+
+/// @todo change this to typed param test
+/// <type, list of string, value pair, list of invalid string, value pair>
+TEST(NameProperty, setPropertyValid_p) {
+  nntrainer::props::Name n;
+  EXPECT_NO_THROW(n.set("layer"));
+  EXPECT_EQ(n.get(), "layer");
```

Comment:
add check with allowed connectors like `_` and `-`.
Also it might better to specify in comments in `name` the allowed characters - alphanumeric + connectors.

Reply:
Updated the testcase